tests/mlmodel_sklearn/test_ml_model.py

```diff
@@ -0,0 +1,323 @@
+            "inference_id": None,
+            "model_name": "PandasTestModel",
+            "model_version": "1.5.0b1",
+            "feature_name": "feature1",
+            "type": "categorical",
+            "value": "5.0",
+        }
+    },
+    {
+        "users": {
+            "inference_id": None,
+            "model_name": "PandasTestModel",
+            "model_version": "1.5.0b1",
+            "feature_name": "feature1",
+            "type": "categorical",
+            "value": "6.0",
+        }
+    },
+    {
+        "users": {
+            "inference_id": None,
+            "model_name": "PandasTestModel",
+            "model_version": "1.5.0b1",
+            "feature_name": "feature2",
+            "type": "categorical",
+            "value": "7.0",
+        }
+    },
+    {
+        "users": {
+            "inference_id": None,
+            "model_name": "PandasTestModel",
+            "model_version": "1.5.0b1",
+            "feature_name": "feature2",
+            "type": "categorical",
+            "value": "8.0",
+        }
+    },
+]
+
+
+@reset_core_stats_engine()
+def test_wrapper_attrs_custom_model_pandas_df():
+    @validate_custom_event_count(count=4)
+    @validate_custom_events(pandas_df_recorded_custom_events)
+    @background_task()
+    def _test():
+        x_train = pandas.DataFrame({"col1": [0, 0], "col2": [1, 1]}, dtype="category")
+        y_train = pandas.DataFrame({"label": [0, 1]}, dtype="category")
+        x_test = pandas.DataFrame({"col1": [5.0, 6.0], "col2": [7.0, 8.0]}, dtype="category")
+
+        model = CustomTestModel().fit(x_train, y_train)
+        wrap_mlmodel(model, name="PandasTestModel", version="1.5.0b1", feature_names=["feature1", "feature2"])
+        labels = model.predict(x_test)
+
+        return model
+
+    _test()
+
+
+pandas_df_recorded_builtin_events = [
+    {
+        "users": {
+            "inference_id": None,
+            "model_name": "MyDecisionTreeClassifier",
+            "model_version": "1.5.0b1",
+            "feature_name": "feature1",
+            "type": "numerical",
+            "value": "6.5",
+        }
+    },
+    {
+        "users": {
+            "inference_id": None,
+            "model_name": "MyDecisionTreeClassifier",
+            "model_version": "1.5.0b1",
+            "feature_name": "feature1",
+            "type": "numerical",
+            "value": "3.4",
+        }
+    },
+    {
+        "users": {
+            "inference_id": None,
+            "model_name": "MyDecisionTreeClassifier",
+            "model_version": "1.5.0b1",
+            "feature_name": "feature2",
+            "type": "numerical",
+            "value": "2.9",
+        }
+    },
+    {
+        "users": {
+            "inference_id": None,
+            "model_name": "MyDecisionTreeClassifier",
+            "model_version": "1.5.0b1",
+            "feature_name": "feature2",
+            "type": "numerical",
+            "value": "7.5",
+        }
+    },
+]
+
+
```

Comment:
We need a couple tests for the error/warning cases:
* if the feature names don't match the number of feature columns and it's a pandas dataframe
* if the feature names don't match the number of feature columns and it's a np array